### PR TITLE
Updated as per Review: CommentController

### LINKED ISSUE #29: [DEV] Comment Controller

### PR DESCRIPTION
**Pull request by pyaeger**

### CODE REVIEW

**Review by FullSol — APPROVED** [review]

File: src/main/java/com/amplifier/controllers/CommentController.java [hunk 1]
```diff
@@ -1,5 +1,74 @@
...
+    
+    
+    @GetMapping("/comments")
+    @ApiOperation(value = "Find all comments", notes = "Pulling all comments from the API", response = Comment.class)
+    public @ResponseBody List<Comment> getAll() {
+        return commentService.getAll();
+    }
+    
+    @PostMapping("/comment")
+    @ApiOperation(value = "Create new comment entity", notes = "Adding a new comment through the API", response = Comment.class)
```
Comment (L3viathon) [inline src/main/java/com/amplifier/controllers/CommentController.java]: `result = Comment.class` in @ApiOperation is not needed for @PostMapping.

`@ApiOperation` describes the operation/HTTP method against a specific path, where `response` specifies the return type of the operation. As a result, you would only need to specify `result = Comment.class` for HTTP get methods (that have something to return) like the two above.

File: src/main/java/com/amplifier/controllers/CommentController.java [hunk 2]
```diff
@@ -1,5 +1,74 @@
...
+    public @ResponseBody ClientMessage createComment(@RequestBody Comment comment) {
+        return commentService.add(comment) ? CREATION_SUCCESSFUL : CREATION_FAILED;
+    }
+
+    @PutMapping("/comment")
+    @ApiOperation(value = "Update comment entity", notes = "Editing a comment through the API", response = Comment.class)
```
Comment (L3viathon) [inline src/main/java/com/amplifier/controllers/CommentController.java]: `result = Comment.class` in @ApiOperation is not needed for @PutMapping.

File: src/main/java/com/amplifier/controllers/CommentController.java [hunk 3]
```diff
@@ -1,5 +1,74 @@
...
+    public @ResponseBody ClientMessage updateComment(@RequestBody Comment comment) {
+        return commentService.edit(comment) ? UPDATE_SUCCESSFUL : UPDATE_FAILED;
+    }
+
+    @DeleteMapping("/comment")
+    @ApiOperation(value = "Remove user entity", notes = "Deleting a comment through the API", response = Comment.class)
```
Comment (L3viathon) [inline src/main/java/com/amplifier/controllers/CommentController.java]: `result = Comment.class` in @ApiOperation is not needed for @DeleteMapping.

File: src/main/java/com/amplifier/controllers/CommentController.java [hunk 4]
```diff
@@ -1,5 +1,74 @@
...
+@Api(value = "CommentRestController", description = "REST controller related to Comment Entities")
+
 public class CommentController {
+	
+    @Autowired
+    private CommentService commentService;
+    
+    @ApiOperation(value = "Find comment by id number", notes = "Provide an id to lookup a specific comment from the API", response = Comment.class)
+    @GetMapping(path = "/comment")
+    public @ResponseBody Comment getById(@RequestParam(value = "id", name = "id") int id) {
```
Comment (L3viathon) [inline src/main/java/com/amplifier/controllers/CommentController.java]: Line 41:

In `(@RequestParam(value = "id", name = "id") int id)`, you only need `(@RequestParam(name = "id") int id)`. 

`name` is the name of the request parameter to bind to, and `value` is just an alias for `name`.

**Review by L3viathon — CHANGES_REQUESTED** [review]
Thanks for doing this! I made a few comments...

**Comment by FullSol** [comment]
> Thanks for doing this! I made a few comments... 

Nice catch on that.

File: src/main/java/com/amplifier/controllers/CommentController.java (hunk 4 above)
Comment (pyaeger) [reply src/main/java/com/amplifier/controllers/CommentController.java]: Ok, making all changes requested. Thx!

**Review by L3viathon — APPROVED** [review]
Awesome, thank you Patrick!

**Pull request merged**